### PR TITLE
Convert wind-gust-diagnostic CLI to use Clize

### PR DESCRIPTION
**Pull request by gavinevans**
Description
This PR converts the `wind_gust_diagnostic` CLI to use Clize.

Testing:
 - [x] Ran tests and they passed OK
 - [x] Added new tests for the new feature(s)

### CODE REVIEW

File: lib/improver/cli/wind_gust_diagnostic.py [hunk 1]
```diff
@@ -31,95 +31,41 @@
...
-    save_netcdf(result, args.output_filepath)
-
-
-def process(gust_cube, speed_cube, percentile_gust, percentile_speed):
+@cli.clizefy
+@cli.with_output
+def process(gust_cube: cli.inputcube,
+            ws_cube: cli.inputcube,
+            *,
+            percentile_gust: float = 50,
```
Comment (arh89) [inline lib/improver/cli/wind_gust_diagnostic.py]:
```suggestion
            percentile_gust: float = 50.0,
```

File: lib/improver/cli/wind_gust_diagnostic.py [hunk 2]
```diff
@@ -31,95 +31,41 @@
...
+            percentile_ws: float = 95):
```
Comment (arh89) [inline lib/improver/cli/wind_gust_diagnostic.py]:
```suggestion
            percentile_ws: float = 95.0):
```

**Comment by arh89** [comment]
Looks good to me - one minor niggle about types, but I'm happy otherwise.

File: lib/improver/cli/wind_gust_diagnostic.py (hunk 1 above)
Comment (gavinevans) [reply lib/improver/cli/wind_gust_diagnostic.py]: I've updated this, as suggested.

File: lib/improver/cli/wind_gust_diagnostic.py (hunk 2 above)
Comment (gavinevans) [reply lib/improver/cli/wind_gust_diagnostic.py]: I've updated this, as suggested.

**Pull request merged**